File: RenetikObjc/Library/Classes/Core/Categories/UIKit/UISearchBar+CSExtension.m
//
// Created by Rene Dohan on 24/01/15.
// Copyright (c) 2015 creative_studio. All rights reserved.
//

#import "UISearchBar+CSExtension.h"


@implementation UISearchBar (CSExtension)

- (void)setTextColor:(UIColor *)textColor {
    for (UIView *subView in self.subviews) {
        for (UIView *secondLevelSubview in subView.subviews) {
            if ([secondLevelSubview isKindOfClass:[UITextField class]]) {
                UITextField *textField = (UITextField *) secondLevelSubview;
                textField.textColor = textColor;
            }
            if ([secondLevelSubview isKindOfClass:[UIButton class]]) {
                UIButton *button = (UIButton *) secondLevelSubview;
                [button setTitleColor:textColor forState:UIControlStateNormal];
                [button setTitleColor:textColor forState:UIControlStateHighlighted];
            }
        }
    }
}


- (void)transparent {
    self.barTintColor = UIColor.clearColor;
    self.backgroundColor = UIColor.clearColor;
}
@end

File: Renetik/Renetik/Classes/Core/Extensions/UIKit/UISearchBar+CSExtension.swift
//
// Created by Rene Dohan on 2/15/20.
//

import UIKit

public extension UISearchBar {

    var textColor: UIColor? {
        get { textFieldInsideSearchBar?.textColor }
        set(value) { textFieldInsideSearchBar?.textColor = value }
    }

    func transparent() {
        barTintColor = UIColor.clear
        backgroundColor = UIColor.clear
    }

    private var textFieldInsideSearchBar: UITextField? { value(forKey: "searchField") as? UITextField }
}